src/main/java/org/apache/maven/plugins/shade/mojo/ShadeMojo.java

```diff
@@ -449,6 +458,24 @@ public void execute()
 
             List<ResourceTransformer> resourceTransformers = getResourceTransformers();
 
+            if ( createDependencyReducedPom )
+            {
+                createDependencyReducedPom( artifactIds );
+            }
+
+            if ( useDependencyReducedPomInJar )
```

Comment:
Maybe this should be nested in the `if ( createDependencyReducedPom )` clause instead of the exception. The documentation can tell that this is a noop if `createDependencyReducedPom ` isn't set?! WDYT?

Reply:
I structured it to give an error in case of an invalid configuration (use but don't create) instead of silently doing a noop.

Reply:
@rfscholte In a few plugin we explicitly ignore params if a condition is not met. Should we apply here the same logic?

Reply:
I agree with @michael-o . It is like calling `clean` when there is nothing to clean. Not worth an exception.

Reply:
Ok, I'll change it.

Reply:
Done.